modules/ocf_printhost/manifests/cups.pp

```diff
@@ -0,0 +1,48 @@
+class ocf_printhost::cups {
+  package { [ 'cups', 'cups-bsd' ]: }
+  ocf::repackage { 'hplip':
+    recommends => false
+  }
+  service { 'cups': }
+
+  File {
+    require => Package['cups'],
+    notify  => Service['cups'],
+  }
+
+  file {
+    '/etc/cups/cupsd.conf':
+      content => template('ocf_printhost/cups/cupsd.conf.erb');
+    '/etc/cups/cups-files.conf':
+      content => template('ocf_printhost/cups/cups-files.conf.erb');
+    '/etc/cups/lpoptions':
+      content => 'Default double';
+    ['/etc/cups/raw.convs', '/etc/cups/raw.types']:
+      content => '# deny printing raw jobs';
+    '/etc/cups/ppd':
+      ensure  => directory,
+      group   => 'lp';
+    ['/etc/cups/ppd/deforestation-double.ppd', '/etc/cups/ppd/logjam-double.ppd']:
+      source  => 'puppet:///modules/ocf_printhost/cups/double.ppd',
+      require => File['/etc/cups/ppd'];
+    ['/etc/cups/ppd/deforestation-single.ppd', '/etc/cups/ppd/logjam-single.ppd']:
+      source  => 'puppet:///modules/ocf_printhost/cups/single.ppd',
+      require => File['/etc/cups/ppd'];
```

Comment:
don't need to require parent directories, puppet automatically sets up the dependency (as long as you actually do declare the parent)

(same comment above)